Python/pystrcmp.c

```diff
@@ -6,21 +6,24 @@
 int
 PyOS_mystrnicmp(const char *s1, const char *s2, Py_ssize_t size)
 {
+    const unsigned char *p1, *p2;
     if (size == 0)
         return 0;
-    while ((--size > 0) &&
-           (tolower((unsigned)*s1) == tolower((unsigned)*s2))) {
-        if (!*s1++ || !*s2++)
+    p1 = (void *)s1;
+    p2 = (void *)s2;
+    for (; (--size > 0) && (tolower(*p1) == tolower(*p2)); p1++, p2++) {
+        if (!*p1 || !*p2)
```

Comment:
Conditions in `for` and `if` can be merged:
```c
(--size > 0) && (tolower(*p1) == tolower(*p2)) && *p1 && *p2
```
It can be rewritten in a form similar to `PyOS_mystricmp()`.
```c
(--size > 0) && *p1 && (tolower(*p1) == tolower(*p2))
```